xtask/src/build_docker_image.rs

```diff
@@ -107,17 +107,25 @@ pub fn build_docker_image(
         if gha && targets.len() > 1 {
             println!("::group::Build {target}");
         }
+        let docker_root = metadata.workspace_root.join("docker");
         let mut docker_build = Command::new(engine);
         docker_build.args(&["buildx", "build"]);
-        docker_build.current_dir(metadata.workspace_root.join("docker"));
+        docker_build.current_dir(&docker_root);
 
         if push {
             docker_build.arg("--push");
         } else {
             docker_build.arg("--load");
         }
 
-        let dockerfile = format!("Dockerfile.{target}");
+        let dockerfile_name = format!("Dockerfile.{target}");
+        let dockerfile_root = if docker_root.join(&dockerfile_name).exists() {
+            docker_root.clone()
+        } else {
+            docker_root.join("cross-toolchains").join("docker")
```

Comment:
seems we're always using the string version of the file, so it could be converted to one here already